### ISSUE TITLE
Bats-core fails to handle a failing test (tracing.bash: line 99: : No such file or directory)

### ISSUE THREAD

**Issue by mat128** [BashSupport-Pro/bashsupport-pro#53]
Hi,

Currently, the bats-core integration fails when running a failing test.

Context:
BashSupport Pro v1.5.9.202
macOS 10.15.7
Happens both under Apple-provided bash 3.2 or brew-installed bash 5.0.
IntelliJ IDEA 2020.2.3 (Ultimate Edition) Build #IU-202.7660.26, built on October 6, 2020

Given a failing test:
```bats
#!/usr/bin/env bats

@test "failing_test" {
    [ "ABC" == "DEF" ]
}
```

When running through official bats 1.2.1:
```console
$ which bats
/usr/local/bin/bats
$ bats --version
Bats 1.2.1
$ bats tests/mytest.bats 
 ✗ failing_test
   (in test file tests/mytest.bats, line 4)
     `[ "ABC" == "DEF" ]' failed

1 test, 1 failure
```

When running through the bashsupport-integrated version, there is a problem in tracing.bash:
![Screen Shot 2020-11-03 at 11 54 26 AM](https://user-images.githubusercontent.com/234168/98017661-90d03f00-1dcd-11eb-9ada-a90aec96fcea.png)
```console
Testing started at 11:54 a.m. ...
/bin/bash "/Users/mmitchell/Library/Application Support/JetBrains/IntelliJIdea2020.2/plugins/bashsupport-pro/bats-core/bin/bats" -t -f ^failing_test$ -r /Users/mmitchell/projects/bashtest/tests/mytest.bats
/Users/mmitchell/Library/Application Support/JetBrains/IntelliJIdea2020.2/plugins/bashsupport-pro/bats-core/lib/bats-core/tracing.bash: line 99: : No such file or directory
   ---
# (from function `' in test file , line 0)

failing_test
```
bash in my case is macOS's bash 3.2:
```console
$ /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)
Copyright (C) 2007 Free Software Foundation, Inc.
```

The same problem occurs when running through bash 5.0.18:
```console
$ /usr/local/bin/bash "/Users/mmitchell/Library/Application Support/JetBrains/IntelliJIdea2020.2/plugins/bashsupport-pro/bats-core/bin/bats" -t -f ^failing_test$ -r /Users/mmitchell/projects/bashtest/tests/mytest.bats
1..1
# _id=1, _file=/Users/mmitchell/projects/bashtest/tests/mytest.bats, _test=failing_test
not ok 1 failing_test
   ---
# (from function `' in test file , line 0)
/Users/mmitchell/Library/Application Support/JetBrains/IntelliJIdea2020.2/plugins/bashsupport-pro/bats-core/lib/bats-core/tracing.bash: line 99: : No such file or directory
```

Reverting tracing.bash to it's original version, by applying the following patch:
```patch
diff -Naur bats-core/lib/bats-core/tracing.bash bats-core.fixed/lib/bats-core/tracing.bash
--- bats-core/lib/bats-core/tracing.bash        2020-11-03 11:52:07.000000000 -0500
+++ bats-core.fixed/lib/bats-core/tracing.bash  2020-11-03 11:52:59.000000000 -0500
@@ -127,8 +127,6 @@
                # Expansion is required for empty arrays which otherwise error
                BATS_CURRENT_STACK_TRACE=("${BATS_STACK_TRACE[@]+"${BATS_STACK_TRACE[@]}"}")
                bats_capture_stack_trace
-       else
-         BATS_CURRENT_STACK_TRACE=("")
        fi
 }
 
```

This resolves the problem under both bash 3.2 and 5.0 while still allowing integration through bashsupport.
![Screen Shot 2020-11-03 at 12 03 40 PM](https://user-images.githubusercontent.com/234168/98017720-a2b1e200-1dcd-11eb-9fe6-dc247031fb82.png)

```console
$ /usr/local/bin/bash "/Users/mmitchell/Library/Application Support/JetBrains/IntelliJIdea2020.2/plugins/bashsupport-pro/bats-core/bin/bats" -t -f ^failing_test$ -r /Users/mmitchell/projects/bashtest/tests/mytest.bats
1..1
# _id=1, _file=/Users/mmitchell/projects/bashtest/tests/mytest.bats, _test=failing_test
not ok 1 failing_test
   ---
# (in test file tests/mytest.bats, line 4)
#   `[ "ABC" == "DEF" ]' failed
   ...
```

I could trace the issue being an empty stack trace, and code assuming to find the filename and line number from top of stack. Bats tracing code assumes the stack trace will always be present. The added else branch reinitializes the stack trace

Completely reverting bats-core to it's upstream version breaks the bashsupport integration, as other parts of the code have been modified.

**Comment by jansorg** [comment]
Thanks for reporting! There‘s CI for failing tests, but obviously not good enough.
The change was for https://github.com/bats-core/bats-core/issues/350. 
I’ll test and fix this for the next bug fix update of 1.5.x.

**Comment by jansorg** [comment]
Fixed for the next update of 1.5, which should be out soon. [The fix](https://github.com/BashSupport-Pro/bats-core/commit/5ba64d20c68518aa7198959997f46182a9211303#diff-66fed6dfb817e19e15a1ac8f9f60ebf22760549535312b5bd33071a1c3749697R61) now does not reinitialize the variable, but checks the length instead.
I'm planning to switch to a custom bats-core test formatter, but only for the next major version.
